pkg/egressgateway/manager_privileged_test.go

```diff
@@ -558,14 +592,7 @@ func parseIPRule(sourceIP, destCIDR, egressIP string, ifaceIndex int) parsedIPRu
 }
 
 func assertIPRules(c *C, rules []ipRule) {
-	var err error
-	for i := 0; i < 10; i++ {
-		if err = tryAssertIPRules(rules); err == nil {
-			break
-		}
-		time.Sleep(10 * time.Millisecond)
-	}
-
+	err := tryAssertIPRules(rules)
```

Comment:
I'm keeping this assert/tryAssert split as it maintains the code more readable imo, as tryAssert just needs to return an error rather than calling c.Assert(err, IsNil) each time